### ISSUE TITLE
Chef service users have interactive logon shells

### ISSUE THREAD

**Issue by trickyearlobe** [chef/chef-server#2150]
# Describe the problem
A large corporate customer scanned their Chef servers with Inspec and found a couple of Chef service accounts with interactive login shells which don't as far as we can tell need interactive login.

To improve security they would like to remove the interactive login shells if they are not required for normal Chef operation.

# Software Version
Chef server 13.2.49 and 14.0.58

# Replication Case
cat `/etc/passwd` and observe there are two users with interactive shells (/bin/sh)

```
opscode:x:997:995::/opt/opscode/embedded:/bin/sh
opscode-pgsql:x:996:994::/var/opt/opscode/postgresql:/bin/sh
```

# Possible Solution
Set the login shell as `/sbin/nologin` for opscode and opscode-pgsql

The defaults seem to be set here
https://github.com/chef/chef-server/blob/d0d773d762486a1e5717e5f05ef074a2151d412e/omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb#L80
https://github.com/chef/chef-server/blob/d0d773d762486a1e5717e5f05ef074a2151d412e/omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb#L540

Aha! Link: https://chef.aha.io/features/SH-425

**Comment by trickyearlobe** [comment]
Related issue in Chef Backend https://github.com/chef/chef-backend/issues/658

**Comment by lbakerchef** [comment]
https://github.com/chef/chef-server/pull/2244

**Comment by jashaik** [comment]
After my analysis, opscode doesn't need interactive login shells but opscode-pgsql needs it in postgresql DB operations.'chef-server-ctl psql App' internally using opscode-pgsql interactive shell for DB operations.
We can add one config for the customers who wants to disable the interactive login shell based on thier DB operations.
Please let me know your view @PrajaktaPurohit @lbakerchef.

**Comment by jashaik** [comment]
As discussed, We are going to remove the interactive shell access for opscode only as opscode-pgsql needs more investigation. @PrajaktaPurohit @lbakerchef @tas50

**Comment by tas50** [comment]
The fix for the `opscode` user is now in place. We weren't able to make the update for the PG user so that's going to take further investigation. This fix will ship in Chef Infra Server 14.2

**Comment by tas50** [comment]
Closing this out since the PG user is required for operations.
